### PR TITLE
fix bug multi tenant auditing, logging and event is incorrect

### LINKED ISSUE #2566: multi tenant auditing is incorrect
**Describe the Bug**
The admin of project, the admin of ws both can't search their own project/ws audit logs. The platform admin and workspace manager can search the audit logs.
Screen Shot 2020-07-23 at 8 23 19 PM
Screen Shot 2020-07-23 at 8 23 37 PM

**Versions Used**
KubeSphere: 3.0.0-dev
Kubernetes: 1.18.6

**Environment**
allinone with auditing enabled

**How To Reproduce**
1. `ray` is the workspace admin, self-provisioner is the admin of the project under the workspace. 
2. Using self-provisioner created several projects.
3. Both `ray` and `self-provisioner` can't search audits. But the platform admin and workspace manager can

**Expected behavior**
ray and self provisioner are able to search their own audits

### LINKED ISSUE #2568: multi-tenant event is incorrect
**Describe the Bug**
The self-provisioner can search its own project events. But the admin of the workspace of the project can't.
Screen Shot 2020-07-23 at 9 22 14 PM
Screen Shot 2020-07-23 at 9 22 37 PM

**Versions Used**
KubeSphere: 3.0.0-dev

### PR DESCRIPTION
**Pull request by wanjunlei**
Signed-off-by: wanjunlei <wanjunlei@yunify.com>

**What type of PR is this?**
/kind bug

**Which issue(s) this PR fixes**:

Fixes #2566 
Fixes #2568

### CODE REVIEW

**Comment by benjaminhuo** [comment]
@wanjunlei pls change the title of this PR if this affects events and logging as well

**Comment by benjaminhuo** [comment]
@wansir @junotx @huanggze pls review this PR

File: pkg/models/tenant/tenant.go [hunk 1]
```diff
@@ -806,21 +805,41 @@ func (t *tenantOperator) Auditing(user user.Info, queryParam *auditingv1alpha1.Q
 	}
 
 	namespaceCreateTimeMap := make(map[string]time.Time)
+
 	for _, ns := range iNamespaces {
-		namespaceCreateTimeMap[ns.Name] = ns.CreationTimestamp.Time
+		listEvts := authorizer.AttributesRecord{
```
Comment (benjaminhuo) [inline pkg/models/tenant/tenant.go:810]: We'd better to add some comments here to elaborate that auditing will use the same authorization logic as events to decide if user could see an auditing event

File: pkg/models/tenant/tenant.go [hunk 2]
```diff
@@ -806,21 +805,41 @@ func (t *tenantOperator) Auditing(user user.Info, queryParam *auditingv1alpha1.Q
...
+			namespaceCreateTimeMap[ns.Name] = ns.CreationTimestamp.Time
+		}
 	}
 	// If there are no ns and ws query conditions,
-	// those events with empty `ObjectRef.Namespace` will also be listed when user can list all namespaces
+	// those events with empty `objectRef.namespace` will also be listed when user can list all events
 	if len(queryParam.WorkspaceFilter) == 0 && len(queryParam.ObjectRefNamespaceFilter) == 0 &&
 		len(queryParam.WorkspaceSearch) == 0 && len(queryParam.ObjectRefNamespaceSearch) == 0 {
-		listNs := authorizer.AttributesRecord{
+		listEvts := authorizer.AttributesRecord{
```
Comment (benjaminhuo) [inline pkg/models/tenant/tenant.go:833]: Same here, We'd better to add some comments here to elaborate that auditing will use the same authorization logic as events to decide if user could see an auditing event

File: pkg/models/tenant/tenant.go [hunk 3]
```diff
@@ -806,21 +805,43 @@ func (t *tenantOperator) Auditing(user user.Info, queryParam *auditingv1alpha1.Q
 	}
 
 	namespaceCreateTimeMap := make(map[string]time.Time)
+
+	// Now auditing and event have the same authority management, so we can determine whether the user
```
Comment (benjaminhuo) [inline pkg/models/tenant/tenant.go:809]: authority management => authorization mechanism

File: pkg/models/tenant/tenant.go [hunk 4]
```diff
@@ -558,11 +558,10 @@ func (t *tenantOperator) DeleteWorkspace(workspace string) error {
...
-// 1. the namespace which belongs to user.
-// 2. the namespace in workspace which is in workspaces when workspaces is not empty.
-// 3. the namespace in workspace which contains one of workspaceSubstrs when workspaceSubstrs is not empty.
-// 4. the namespace which is in namespaces when namespaces is not empty.
-// 5. the namespace which contains one of namespaceSubstrs when namespaceSubstrs is not empty.
+// 1. the namespace in workspace which is in workspaces when workspaces is not empty.
+// 2. the namespace in workspace which contains one of workspaceSubstrs when workspaceSubstrs is not empty.
+// 3. the namespace which is in namespaces when namespaces is not empty.
+// 4. the namespace which contains one of namespaceSubstrs when namespaceSubstrs is not empty.
 func (t *tenantOperator) listIntersectedNamespaces(user user.Info,
```
Comment (huanggze) [inline pkg/models/tenant/tenant.go:565]: So now `user user.Info` is an unnecessary parameter, and cen be removed.

**Pull request merged**